pandera/schema_components.py

```diff
@@ -202,10 +202,10 @@ def validate_column(check_obj, column_name):
         )
 
         for column_name in column_keys_to_check:
-            if self.coerce:
-                check_obj[column_name] = self.coerce_dtype(
-                    check_obj[column_name]
-                )
+            # if self.coerce:
```

Comment:
this won't work because it disables column coercion when `Column` objects are [used in isolation](https://pandera.readthedocs.io/en/stable/dataframe_schemas.html#stand-alone-column-validation).

The fix for this issue should actually be right here: https://github.com/unionai-oss/pandera/blob/master/pandera/schemas.py#L681-L691

Since the `DataFrameSchema.coerce_dtype` is called right above this section, we can disable column-level coercion when applying checks like so:

```python
        schema_components = []
        for col_name, col in self.columns.items():
            if (
                col.required or col_name in check_obj
            ) and col_name not in lazy_exclude_columns:
                # the column-level coerce option the dataframe schema
                # shouldn't take effect for component validation
                col = copy.deepcopy(col)
                col._coerce = False
                if self.dtype is not None:
                    # override column dtype with dataframe dtype
                    col.dtype = self.dtype
                schema_components.append(col)
```

This way, coercion will only occur once (at the dataframeschema-level) and not when the column components are used in line `704`

Reply:
implemented